File: exam_in_class/3/document_counting.py
# author: jamie
# email: jinjiedeng.jjd@gmail.com
if __name__ == "__main__":
    n = int(input())
    inverted_index = {}
    for i in range(n):
        m_i = int(input())
        document = input().split()
        for word in document:
            if word not in inverted_index:
                inverted_index[word] = set()
            inverted_index[word].add(i)
    q = int(input())
    for _ in range(q):
        word = input()
        print(0 if word not in inverted_index else len(inverted_index[word]))

File: exam_in_class/3/document_counting.cc
/*
* author: jamie
* email: jinjiedeng.jjd@gmail.com
*/
#include <iostream>
#include <map>
#include <set>

int main ()
{
  uint64_t n; std::cin >> n;
  std::map<std::string, std::set<uint64_t>> inverted_index;
  for (uint64_t i {}; i != n; ++i)
  {
    uint64_t m_i; std::cin >> m_i;
    while (m_i--)
    {
      std::string word; std::cin >> word;
      inverted_index[word].insert(i);
    }
  }
  uint64_t q; std::cin >> q;
  while (q--)
  {
    std::string word; std::cin >> word;
    std::cout << inverted_index[word].size() << "\n";
  }
  return 0;
}